src/routes/RecipeDetails.js

```diff
@@ -1,12 +1,42 @@
 import React from 'react'
 import PropTypes from 'prop-types'
 
-const RecipeDetails = props => {
-  const { id } = props.match.params
-  
-  return (
-    <h1 className="text-center">{id}</h1>
-  )
+class RecipeDetails extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      recipeDetails: {},
+    }
+  }
+
+  async componentDidMount() {
+    const response = await fetch(`/api/recipes/${this.props.match.params.id}.json`)
+    const recipeDetails = await response.json()
+    this.setState({ recipeDetails })
+  }
+
+  render() {
+    const { recipeDetails } = this.state
+    const recipeInfo = recipeDetails.recipeDetail
```

Comment:
I think your names are a bit confusing. It would be better to call it `recipe` and `recipeDetail`
You can destructure both if you want.
```js
const { recipe, recipe: { recipeDetail } } = this.state;

// 👆 this is the same as 👇 

const { recipe } = this.state;
const { recipeDetail } = recipe;
```